org.eclipse.scout.sdk.s2i/src/main/kotlin/org/eclipse/scout/sdk/s2i/util/SourceFolderHelper.kt

```diff
@@ -106,122 +98,210 @@ class SourceFolderHelper(val project: Project, val sourceFolder: SourceFolder, v
                 .filter { !isGeneratedSourceFolder(it) }
                 .map { it to Strings.levenshteinDistance(it.contentEntry.rootModel.module.name, reference.name) }
                 .sortedBy { it.second }
-                .firstOrNull { (sf, _) -> myScoutTierOfModule(sf.contentEntry.rootModel.module) == scoutTier }?.first ?: return null
+                .firstOrNull { (sf, _) -> myTierOfModule(sf.contentEntry.rootModel.module) == tier }?.first ?: return null
```

Comment:
Use tier.isIncludedIn(myTierOfModule(sf.contentEntry.rootModel.module)) instead of ==